File: mifosng-android/src/main/java/com/mifos/utils/services/data/MifosGroup.java
package com.mifos.utils.services.data;

import com.google.gson.Gson;
import com.mifos.objects.db.Level;
import com.mifos.objects.db.Staff;

public class MifosGroup {
    public int groupId;
    public String groupName;
    public int staffId;
    public String staffName;

    public int levelId;
    public String levelName;

    public Client[] clients;


    @Override
    public String toString() {
        return new Gson().toJson(this);
    }

    public com.mifos.objects.db.MifosGroup getData() {
        com.mifos.objects.db.MifosGroup group = new com.mifos.objects.db.MifosGroup();
        group.setGroupId(this.groupId);
        group.setGroupName(this.groupName);

        Level level = new Level();
        level.setLevelId(this.levelId);
        level.setLevelName(this.levelName);
        level.setGroupId(group.getGroupId());

        Staff staff = new Staff();
        staff.setStaffId(this.staffId);
        staff.setStaffName(this.staffName);
        staff.setGroupId(group.getGroupId());

        group.setStaff(staff);
        group.setLevel(level);

        return group;
    }

}

File: core/model/src/commonMain/kotlin/com/mifos/core/model/objects/databaseobjects/MifosGroup.kt
/*
 * Copyright 2024 Mifos Initiative
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 *
 * See https://github.com/openMF/android-client/blob/master/LICENSE.md
 */
package com.mifos.core.model.objects.databaseobjects

import com.mifos.core.common.utils.Parcelable
import com.mifos.core.common.utils.Parcelize
import kotlinx.serialization.Serializable

@Parcelize
@Serializable
data class MifosGroup(
    var staffId: Int = 0,

    var staffName: String? = null,

    var levelId: Int = 0,

    var levelName: String? = null,

    var groupId: Long = 0,

    var groupName: String? = null,

    var centerId: Long = 0,

    var clients: List<Client> = ArrayList(),
) : Parcelable
